.github/workflows/SpellCheck.yml

```diff
@@ -0,0 +1,17 @@
+name: Spell Check
+
+permissions: {}
```

Comment:
```suggestion
permissions:
  contents: read
```

Reply:
`permissions: {}` is strictly less permissive (rmissions-to-jobs#defining-access-for-the-github_token-scopes). Why add these additional permissions?

Reply:
Yeah, let's change it back to `permissions: {}`.

I'm not 100% sure how `actions/checkout` is able to get the contents of the repo even when we set `permissions: {}`. I wonder if the `permissions` key applies only to uses of the GitHub API (and doesn't apply to access over Git SSH).

Reply:
The repo is public, they could be downloading with https.jl for all I know, lol.

Reply:
😂 

For the API, GitHub rate-limits`https://api.github.com/*`, and the rate-limits are much lower if you don't provide a GitHub token.

I'm guessing that GitHub also has some rate-limiting or throttling in place for `https://github.com/*`.